pkg/build/strategies/sti/sti.go

```diff
@@ -560,7 +560,6 @@ func (builder *STI) Execute(command string, user string, config *api.Config) err
 		}()
 
 		opts.Stdin = r
-		defer wg.Wait()
```

Comment:
This is a very tricky defer statement, there is no clean way to "cancel" it.

Calling `wg.Done()` earlier introduces a concurrency problem. As @bparees and @gabemontero noted, `wg.Done()` could end up being called multiple times sending the counter below 0, causing a panic.

Correct code arranges to call `wg.Done()` as many times as `wg.Add(n)` in all cases.